project/CassandraSparkBuild.scala

```diff
@@ -41,20 +48,25 @@ object CassandraSparkBuild extends Build {
     ).copy(dependencies = Seq(embedded % "test->test;it->it,test;")
   ) configs (IntegrationTest, ClusterIntegrationTest)
 
-  lazy val jconnector = Project(
-    id = s"$namespace-java",
-    base = file(s"$namespace-java"),
-    settings = connector.settings,
-    dependencies = Seq(connector % "compile;runtime->runtime;test->test;it->it,test;provided->provided")
+  lazy val jconnector = CrossScalaVersionsProject(
+    name = s"$namespace-java",
+    conf = assembledSettings ++ Seq(libraryDependencies ++= Dependencies.connector ++ Seq(
+        "org.scala-lang" % "scala-reflect"  % scalaVersion.value,
+        "org.scala-lang" % "scala-compiler" % scalaVersion.value % "test,it"), compileOrder in Compile := CompileOrder.ScalaThenJava)
+    ).copy(dependencies = Seq(connector % "compile;runtime->runtime;test->test;it->it,test;provided->provided")
   ) configs (IntegrationTest, ClusterIntegrationTest)
 
   lazy val demos = RootProject("demos", demosPath, Seq(simpleDemos, kafkaStreaming, twitterStreaming))
 
-  lazy val simpleDemos = Project(
-    id = "simple-demos",
-    base = demosPath / "simple-demos",
-    settings = demoSettings,
-    dependencies = Seq(connector, jconnector, embedded)
+  lazy val simpleDemos =
+    Project(
```

Comment:
16:57:05 ➜  ~/repos/spark-cassandra-connector git:(SPARKC-95) ✗ sbt simple-demos/run
   [info] Loading project definition from /Users/russellspitzer/repos/spark-cassandra-connector/project
    
     Scala: 2.10.4 [To build against Scala 2.11 use '-Dscala-2.11=true']
     Scala Binary: 2.10
     Java: target=1.7 user=1.7.0_60

    [info] Set current project to root (in build file:/Users/russellspitzer/repos/spark-cassandra-connector/)
    [info] Compiling 1 Java source to /Users/russellspitzer/repos/spark-cassandra-connector/spark-cassandra-     connector-demos/simple-demos/target/scala-2.10/classes...
    [warn] Multiple main classes detected.  Run 'show discoveredMainClasses' to see the list

     Multiple main classes detected, select one to run:
     [1] com.datastax.spark.connector.demo.WordCountDemo
     [2] com.datastax.spark.connector.demo.TableCopyDemo
     [3] com.datastax.spark.connector.demo.AkkaStreamingDemo
     [4] com.datastax.spark.connector.demo.SQLDemo
     [5] com.datastax.spark.connector.demo.BasicReadWriteDemo